test/tests/functional/pbs_job_array.py

```diff
@@ -39,12 +39,50 @@
 
 
 from tests.functional import *
+from ptl.utils.pbs_logutils import PBSLogUtils
 
 
 class TestJobArray(TestFunctional):
     """
     Test suite for job array feature
     """
+    lu = PBSLogUtils()
+    qjh = """
+import pbs
+
+e = pbs.event()
+j = e.job
+if j.max_run_subjobs is None:
+    j.max_run_subjobs = %d
+pbs.logmsg(pbs.LOG_DEBUG, "max_run_subjobs set to %%d" %% j.max_run_subjobs)
+e.accept()
+"""
+    mjh = """
+import pbs
+
+e = pbs.event()
+j = e.job
+if j.max_run_subjobs != 0:
+    if j.max_run_subjobs > 10:
+        j.max_run_subjobs = %d
+pbs.logmsg(pbs.LOG_DEBUG, "max_run_subjobs set to %%d" %% j.max_run_subjobs)
+e.accept()
+"""
+
+    def create_hook(self, max_run, event, name):
```

Comment:
Maybe make this function name less generic?  This is a generic test plan.  Something like create_max_run_subjobs_hook() ?